sql-plugin/src/main/scala/org/apache/spark/sql/catalyst/json/rapids/GpuJsonScan.scala

```diff
@@ -267,7 +267,14 @@ object JsonPartitionReader {
       RmmRapidsRetryIterator.withRetryNoSplit(dataBufferer.getBufferAndRelease) { dataBuffer =>
         withResource(new NvtxWithMetrics(formatName + " decode",
           NvtxColor.DARK_GREEN, decodeTime)) { _ =>
-          Table.readJSON(cudfSchema, jsonOpts, dataBuffer, 0, dataSize)
+          try {
+            Table.readJSON(cudfSchema, jsonOpts, dataBuffer, 0, dataSize)
+          } catch {
+            case e: AssertionError if e.getMessage == "CudfColumns can't be null or empty" =>
```

Comment:
Parsing exception messages is an anti-pattern.  Would be nice to have a followup to get a more specific exception type to help distinguish this case.

Reply:
I agree, probably would be better to do this for all Exceptions and AssertionErrors.

Reply:
Thanks. I filed https://github.com/rapidsai/cudf/issues/14331